packages/models-library/src/models_library/services_resources.py

```diff
@@ -1,10 +1,36 @@
-from typing import Union
+import logging
+from typing import Any, Final, Union
 
-from models_library.generics import DictModel
-from pydantic import BaseModel, StrictFloat, StrictInt, root_validator
+from pydantic import (
+    BaseModel,
+    ByteSize,
+    Field,
+    StrictFloat,
+    StrictInt,
+    constr,
+    parse_obj_as,
+    root_validator,
+)
 
+from .utils.fastapi_encoders import jsonable_encoder
+
+logger = logging.getLogger(__name__)
+
+ComposeImage = constr(regex=r"[\w/-]+:[\w.@]+")
+DockerComposeServiceName = constr(regex=r"^[a-zA-Z0-9._-]+$")
 ResourceName = str
 
+DEFAULT_SINGLE_SERVICE_NAME: Final[DockerComposeServiceName] = "container"
+
+_MB: Final[int] = 1024 * 1024
+MEMORY_50MB: Final[int] = 50 * _MB
+MEMORY_250MB: Final[int] = 50 * _MB
+MEMORY_1GB: Final[int] = 1024 * _MB
```

Comment:
why not just using ```parse_obj_as(ByteSize, "1Gib")```?

Reply:
and these constants are used only in proxy.py in director-v2, I would move them over. to have these concerns together.

Reply:
Thanks for making me check this. I've forgotten to enforce the minimum memory limit on the `dynamic-sidecar`. Not set to 1 GB similar to the cpu limit set to 1.0 CPU.

I've just placed these values here to be used as constants (prior request). We have all definitions of resources unified.

Reply:
well then... it seems MB now means power of 10s... and MiB is power of 2.
see [https://en.wikipedia.org/wiki/Byte#Multiple-byte_units](https://en.wikipedia.org/wiki/Byte#Multiple-byte_units)

Reply:
It is also encoded in pedantic very clearly. Went for `mib` and `gib`.